src/mscorlib/src/System/Reflection/Emit/ILGenerator.cs

```diff
@@ -1174,6 +1174,25 @@ public virtual void ThrowException(Type excType)
             this.Emit(OpCodes.Throw);
         }
 
+        private static Type GetConsoleType()
+        {
+#if PLATFORM_UNIX
+            Type t = null;
+            try
+            {
+                t = Type.GetType("System.Console, System.Console, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", throwOnError: false);
+            }
+            catch { } // even with throwOnError:false, could still throw for a variety of reasons
```

Comment:
I was just going by the MSDN documentation at https://msdn.microsoft.com/en-us/library/c5cf8k43(v=vs.110).aspx which state that even for throwOnError:false a FileLoadException or BadImageFormatException can be thrown.  If the docs are out of date, or if we believe those exceptions should just propagate instead of translating them into a PlatformNotSupportedException, I'm fine removing the try/catch.  I'll do so unless I hear otherwise from you.